### PR TITLE
#681 convert console logs to errors

### LINKED ISSUE #681: [igEditors/Notifier] Convert console logs to errors for incorrect options
The editors and notifier use some console logs, which should be brought in line with the Errors other controls throw in such cases.
Since that is something intended for the developers - it might even be preferable for wrong initialization to be an error to attract attention to whatever value was not acceptable.

### PR DESCRIPTION
**Pull request by IvayloG**
Closes #681   .  

Unify the exceptions throwing mechanism. Use regular JS error object, so the error is displayed consistently trough browsers.

### CODE REVIEW

File: tests/unit/editors/dateEditor/tests.html [hunk 1]
```diff
@@ -68,7 +68,7 @@
 						value: '01/09/50',
 						dateInputFormat: "dd/MM/yyyy",
 						dateDisplayFormat: "dd/MM/yyyy",
-						centuryThreshold: 101
+						//centuryThreshold: 101
```
Comment (kdinev) [inline tests/unit/editors/dateEditor/tests.html]: Why is this commented out?

Comment (IvayloG) [inline tests/unit/editors/dateEditor/tests.html]: There is a new test created in order to test the centuryThreshold option, and it is removed here as it is no longer relevant for the tests using this editor instance.

**Comment by ig-ealbert** [comment]
Translated messages to Japanese.  Waiting for review.  Once they are reviewed (Friday), how would you like me to add them to the .ja.js file?  Should I send them to you by email so you can add them, or submit a pull request? @Lipata @kdinev @IvayloG

**Comment by IvayloG** [comment]
@ig-ealbert Please add the necessary changes via pull request.

**Comment by ig-ealbert** [comment]
notSupportedState: "サポートされていない通知状態です。success、info、warning、error のいずれかのサポートされる状態を使用してください。",
notSupportedMode: "サポートされていない通知モードです。auto、popover、inline のいずれかのサポートされるモードを使用してください。"

**Pull request merged**